docs/pnp/migration-guides/modal-v7-to-v8.mdx

```diff
@@ -0,0 +1,170 @@
+---
+title: PnP Modal SDK - v7 to v8
+displayed_sidebar: docs
+description: "PnP Modal SDK - v7 to v8 | Documentation - Web3Auth"
+sidebar_label: v7 to v8
+---
+
+# Migration Guide from v7 to v8 for Web3Auth Modal SDK
+
+## General
+
+### `CustomChainConfig`
+
+#### `blockExplorer` is now `blockExplorerUrl`
+With v8, in `CustomChainConfig` accepts `blockExplorerUrl` instead of `blockExplorer`.
+
+#### A new `logo` parameter is now added to CustomChainConfig
+
+In `CustomChainConfig`, a new parameter `logo` has been introduced which can be used to pass
+the logo for the chain.
+
+```tsx
+const chainConfig = {
+    chainId: "0x3",
+    displayName: "Solana Testnet",
+    chainNamespace: CHAIN_NAMESPACES.SOLANA,
+    tickerName: "SOLANA",
+    ticker: "SOL",
+    decimals: 18,
+    rpcTarget: "https://api.testnet.solana.com",
+    // highlight-start
+    blockExplorerUrl: "https://explorer.solana.com/?cluster=testnet",
+    logo: "https://images.toruswallet.io/sol.svg"
+    // highlight-end
+};
+```
+
+### `Web3AuthOptions`
+
+#### A new mandatory `privateKeyProvider` parameter is now added to `Web3AuthOptions`
+
+With v8, it's mandatory to provide `privateKeyProvider` in Web3AuthOptions for your corresponding chain namespace. To know more in-depth about providers,
+have a look at your [Providers reference](/sdk/helper-sdks/providers).
+
+#### New language support for `defaultLanguage`
+
+With v8, you can now use Turkish as a `defaultLanguage` option in `uiConfig`.
+
+```tsx
+const chainConfig = {
+  chainId: "0x1", // Please use 0x1 for Mainnet
+  rpcTarget: "https://rpc.ankr.com/eth",
+  displayName: "Ethereum Mainnet",
+  blockExplorerUrl: "https://etherscan.io/",
+  ticker: "ETH",
+  tickerName: "Ethereum",
+  logo: "https://images.toruswallet.io/eth.svg",
+};
+
+// highlight-start
+const ethereumPrivateKeyProvider = EthereumPrivateKeyProvider({
+  config: {chainConfig}
+});
+// highlight-end
+
+const web3auth = new Web3Auth({
+  clientId: "", // Get your Client ID from the Web3Auth Dashboard
+  web3AuthNetwork: "sapphire_mainnet",
+  chainConfig: {
+    ...chainConfig,
+    chainNamespace: CHAIN_NAMESPACES.EIP155,
+  },
+  uiConfig: {
+    // highlight-next-line
+    defaultLanguage: "tr", // en, de, ja, ko, zh, es, fr, pt, nl
```

Comment:
You can change text “en” to type.
import { LANGUAGES } from "@toruslabs/openlogin-utils";
LANGUAGES.tr

Reply:
Done